src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/EventTypeTest.java

```diff
@@ -0,0 +1,135 @@
+import org.junit.Test;
+import uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.AssignApplicantSolicitorHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.CallbackHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackRequest;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.GeneralApplicationHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.intervener.IntervenerHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Slf4j
+public class EventTypeTest {
+
+    private final List<Class> handlerClassesToIgnore = Arrays.asList(FinremCallbackHandler.class, FinremCallbackRequest.class,
+        GeneralApplicationHandler.class, CallbackHandler.class, IntervenerHandler.class, AssignApplicantSolicitorHandler.class);
+
+    @Test
+    public void shouldOnlyBeOneEventHandlerPerEventType() throws ClassNotFoundException {
+        var handlerClasses = getHandlerClasses();
+        List<EventType> eventTypeList = new ArrayList<>(Arrays.stream(EventType.values()).toList());
+        List<CaseType> caseTypeList = new ArrayList<>(Arrays.stream(CaseType.values()).toList());
+        List<CallbackType> callbackTypeList = new ArrayList<>(Arrays.stream(CallbackType.values()).toList());
+        for (EventType event : eventTypeList) {
+            for (CaseType caseType : caseTypeList) {
+                for (CallbackType callbackType : callbackTypeList) {
+                    List<Class> matchingClasses = new ArrayList<>();
+                    handlerClasses.forEach(clazz -> {
+                        try {
+                            if (!handlerClassesToIgnore.contains(clazz)) {
+                                if (invokeCanHandleMethod(clazz, callbackType, caseType, event)) {
+                                    matchingClasses.add(clazz);
+                                }
+                            }
+                        } catch (InvocationTargetException | InstantiationException | IllegalAccessException
+                                 | NoSuchMethodException ex) {
+                            throw new RuntimeException(ex);
+                        }
+                    });
+                    if (matchingClasses.size() > 1) {
+                        throw new RuntimeException("The combination of event type " + event + ", case type " + caseType
+                            + " and callback type " + callbackType + " has been found in more than one class " + matchingClasses);
+                    }
+                }
+            }
+        }
```

Comment:
where is the assertion for the test method? there should be an `assertThat()` or as a minimum a `verify()` depending on the test circumstances